homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -35,22 +46,149 @@ class PassiveBluetoothEntityKey:
 
 _T = TypeVar("_T")
 
+STORAGE_KEY = "bluetooth.passive_update_processor"
+PASSIVE_UPDATE_PROCESSOR = "passive_update_processor"
+
+
+@dataclasses.dataclass(slots=True, frozen=False)
+class PassiveBluetoothProcessorData:
+    """Data for the passive bluetooth processor."""
+
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = dataclasses.field(
+        default_factory=set
+    )
+    restore_data: dict[str, dict[str, dict[str, Any]]] = dataclasses.field(
+        default_factory=dict
+    )
+
+
+class PassiveBluetoothEntityKeyDict(TypedDict):
+    """Passive bluetooth entity key dict."""
+
+    key: str
+    device_id: str | None
+
+
+class RestoredData(TypedDict):
+    """Restored data."""
+
+    devices: dict[str | None, DeviceInfo]
+    entity_descriptions: dict[PassiveBluetoothEntityKeyDict, dict[str, Any]]
+    entity_names: dict[PassiveBluetoothEntityKeyDict, str | None]
+    entity_data: dict[PassiveBluetoothEntityKeyDict, Any]
+
 
 @dataclasses.dataclass(slots=True, frozen=True)
 class PassiveBluetoothDataUpdate(Generic[_T]):
     """Generic bluetooth data."""
 
     devices: dict[str | None, DeviceInfo] = dataclasses.field(default_factory=dict)
-    entity_descriptions: Mapping[
+    entity_descriptions: dict[
         PassiveBluetoothEntityKey, EntityDescription
```

Comment:
Entity description is the one for the platform so it should be a type var

Reply:
That would be nice but should be another PR since we don't need it right now and likely requiring that is a breaking change